### ISSUE TITLE
hook up dynamic/adaptive worker selection/rejection model to the persistence/db model

### ISSUE THREAD

**Issue by Prtfw** [OpenMined/PyGrid#550]
**Is your feature request related to a problem? Please describe.**
this function here: https://github.com/OpenMined/PyGrid/blob/dev/grid/app/main/routes/federated.py#L271

implements a probabilistic / adaptive random selection of workers who requests to join a federated learning (FL) cycle (approx equal to 1 epoch of training on local data) 

this prototype model aims to leverage randomness to improve adversarial robustness to model poisoning attacks and also aims to maximize the rate of a successful FL cycle (we need enough workers to join each FL cycle so there are enough model parameter diffs / local gradients updates to average + other logistical constraints... i.e. worker must satisfy certain connection speed requirements etc...)

**Describe the solution you'd like**
because this model was developed/implemented prior to a solidified persistence/db schema a number of key variables are currently "hard coded" 

the mission (should you choose to accept... this issue won't self destruct :trollface: ): 
work as a research engineer and
- connect the variables that is related to server_config
- implement tracking of requests in sql alchemy to remove hard coded variables such as `normalized_lambda_actual` etc... stated here: https://github.com/OpenMined/PyGrid/blob/dev/app/main/routes/federated.py#L417
- setup a simulation to test / set appropriate priors / hyper-parameters for the model (this will be a separate issue... will be linked when issue is "written up" (see partially implemented tests + reference implementation in google colab below)

reference:
https://github.com/OpenMined/Roadmap/blob/master/web_and_mobile_team/projects/federated_learning.md (specifically part 2: test)

original pr: https://github.com/OpenMined/PyGrid/pull/479

reference / prototype implementation in colab: https://colab.research.google.com/drive/1u1wIU9IXwkQC-K6Zx7K8iS65kqjT8lQO

**Describe alternatives you've considered**
This model must be connected to persistence and tested... no other alternatives : )

**Additional context**
you will be working with @Prtfw closely and will have support from the pygrid core team.
we would like to have this ticket completed by end of may / mid june.
I am happy to discuss this issue on a call to give you more context / details.

**Next phase of the project:** 
What I hav in mind:
research => select workers based on their predicted fitness for participation in successful training in a privacy preserving and secure way 

participation on this issue will help you ramp up / get to a strong position to participate / contribute in the next phase.

**Comment by cereallarceny** [comment]
Closing in favor of #589
